typing/typecore.ml

```diff
@@ -2714,7 +2715,27 @@ let remaining_function_type ty_ret mode_ret rev_args =
   in
   ty_ret
 
-let collect_unknown_apply_args env funct ty_fun mode_fun rev_args sargs ret_tvar =
+let check_local_application_complete ~env ~app_loc ~sarg ~rest ~mode_fun ~mode_arg ~mode_ret =
+  let submode m1 m2 =
+    match Alloc_mode.submode m1 m2 with
+    | Ok () -> ()
+    | Error () ->
+      let loc =
+        Location.{ loc_start = app_loc.loc_start;
+                   loc_end = sarg.pexp_loc.loc_end;
+                   loc_ghost = app_loc.loc_ghost || sarg.pexp_loc.loc_ghost }
```

Comment:
I think this location may be confusing in the case where the arguments have been passed out of order. I think that sometimes there will be no single span that corresponds to the complete application, so we'll need to think of something else to do there.